snuba/subscriptions/executor_consumer.py

```diff
@@ -326,6 +334,14 @@ def submit(self, message: Message[KafkaPayload]) -> None:
             subscription_id_to_float(subscription_id) < executor_sample_rate
         )
 
+        # Don't execute stale subscriptions
+        if (
+            self.__stale_threshold_seconds is not None
+            and time.time() - datetime.timestamp(task.timestamp)
+            >= self.__stale_threshold_seconds
+        ):
+            should_execute = False
```

Comment:
THis would not commit though while you churn the backlog. Is that correct and intentional?

Reply:
Well, it was somewhat intentional in that I didn't want to have to bring the commit function over to the ExecuteQuery just for this- committing usually happens in the following step. My thinking was that if no query is actually being executed, the backlog can be churned so fast by the executor that in practice not committing until we see a recent timestamp will never really be an issue.

Reply:
Now that you would be introducing the commit function via https://github.com/getsentry/snuba/pull/2656/files#diff-b81082d36647653087818cb560a918148759eb0e770ef1c04313589ea32167d7R196 to this processing step, you have an option to actually perform a commit in this step. Not saying that you need to since you mentioned that it's a temporary code in the other PR.